atc/atccmd/command.go

```diff
@@ -116,6 +116,7 @@ type RunCommand struct {
 	ResourceTypeCheckingInterval time.Duration `long:"resource-type-checking-interval" default:"1m" description:"Interval on which to check for new versions of resource types."`
 
 	ContainerPlacementStrategy        string        `long:"container-placement-strategy" default:"volume-locality" choice:"volume-locality" choice:"random" choice:"fewest-build-containers" description:"Method by which a worker is selected during container placement."`
+	MaxBuildTasksWorker               int           `long:"max-build-tasks-worker" default:"0" description:"Maximum allowed number of active build tasks per worker. Has effect only when used with fewest-build-containers placement strategy. 0 means no limit."`
```

Comment:
Is better explained in the comments, but just for reference: while creating this PR I've realized that `build_containers` is not a reliable number to base the kind of scheduling I'm trying to achieve (as containers stays around long after the workers completed processing).

Reply:
Hey @aledeganopix4d 
To better understand the issue with not being able to use `build_containers` and tracking only `task` containers, do you mean resource `get` & `put` containers are staying around much longer after a build is completed ?

Reply:
Hi @xtreme-sameer-vohra, no, that's not the case.
`build_containers` and `task containers` are actually the same thing: e.g. the container where Tasks are executed. However, the lifetime of a `build_container` is longer than the time the worker is **actively** running the task, this is because the `build_container` will stay on the worker until Garbage Collected which will happen:

1. After ~1 minute if the task was successful
2. After 1 hour if the task errored or failed
While the first case wouldn't be that bad, the second really throws a wrench into any scheduling system that relies on counting the build_containers to **limit** the amount of tasks on a worker.

Reply:
Thanks for the clarification. I understand what you mean now.

Internally within Concourse, the hierarchy is; 
- `check` containers 
- `build` containers
    * `get`
    * `put`
    * `task` ( successful & error'ed )
    * hook containers ( on_failure, on_success , etc )

Adding workload limiting at the `build` container level prior to adding it at the sub `task` level allows `Concourse` as a system to remain general purpose and then providing knobs to be optimal in specific cases.

Reply:
Just a quick clarification, `build_containers` include `get`, `put` and `task` containers, basically just excluding check containers which are assumed to be fairly lightweight.

Reply:
Ah! Sorry, that was indeed a misunderstanding on my part!